### ISSUE TITLE
Upgrading from `3.2.10` to `3.6.8` requires adding `quarkus-reactive-routes`

### ISSUE THREAD

**Issue by fbuetler** [quarkusio/quarkus#38475]
### Describe the bug

We recently upgrade our quarkus application from `3.2.10` to `3.6.8` by using `quarkus upgrade` and following the various migration guides:

* [3.3.0](https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.3)
* [3.4.1](https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.4)
* [3.5.0](https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.5)
* [3.6.0](https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.6)

After the upgrade, compilation failed because some imports from vertx-web were not found (in our case `io.quarkus.vertx.web.RouteFilter` and `io.quarkus.vertx.web.Body`). 

Adding the `quarkus-reactive-routes` dependency resolved the issue.

However, neither of the migrations guides mentioned anything about this missing dependency. So I guess something slipped through.

### Expected behavior

That the imported classes from `io.quarkus.vertx.web` were found.

### Actual behavior

The imported class from `io.quarkus.vertx.web` were not found.

### Output of `uname -a` or `ver`

Linux Colter 5.10.0-1034-oem #35-Ubuntu SMP Wed Jun 23 09:55:56 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk 17.0.9 2023-10-17 
OpenJDK Runtime Environment (build 17.0.9+9-Ubuntu-122.04)
OpenJDK 64-Bit Server VM (build 17.0.9+9-Ubuntu-122.04, mixed mode, sharing)

### Quarkus version or git rev

3.6.8

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.5 (57804ffe001d7215b5e7bcb531cf83df38f93546) 
Maven home: /home/fbuetler/.mvn 
Java version: 17.0.9, vendor: Private Build, runtime: /usr/lib/jvm/java-17-openjdk-amd64 
Default locale: en_GB, platform encoding: UTF-8 
OS name: "linux", version: "5.10.0-1034-oem", arch: "amd64", family: "unix"

**Comment by gsmet** [comment]
Thanks for the report. Could you clarify which components need these classes because I don't think that's expected?

**Comment by fbuetler** [comment]
These symbols were not missing in any component, but we used them in our application code. I just wanted to point out, that using `quarkus-reactive-routes` as a dependency was not necessary before the upgrade. Hence, I think these symbols somehow were available in `3.2.7`, but are not longer available in `3.6.8` without `quarkus-reactive-routes`.

**Comment by gsmet** [comment]
Oh. Any chance you could point which extension was bringing `quarkus-reactive-routes` for you before? I will document it in the migration guide.

Also we might be able to provide an update recipe for `quarkus update`.

**Comment by fbuetler** [comment]
I stripped everything unrelated. Apparently, `quarkus-smallrye-jwt` included `quarkus-reactive-routes` in `3.2.10`, but is not included anymore in `3.6.8`:

`3.2.10`:
```
[INFO] +- io.quarkus:quarkus-smallrye-jwt:jar:3.2.10.Final:compile
[INFO] |  +- io.smallrye:smallrye-jwt:jar:4.3.0:compile
[INFO] |  +- jakarta.annotation:jakarta.annotation-api:jar:2.1.1:compile
[INFO] |  +- io.quarkus:quarkus-reactive-routes:jar:3.2.10.Final:compile
[INFO] |  |  \- io.smallrye.common:smallrye-common-annotation:jar:2.1.0:compile
[INFO] |  +- io.quarkus:quarkus-security:jar:3.2.10.Final:compile
[INFO] |  |  \- jakarta.interceptor:jakarta.interceptor-api:jar:2.1.0:compile
[INFO] |  \- io.quarkus:quarkus-jsonp:jar:3.2.10.Final:compile
[INFO] |     \- org.eclipse.parsson:parsson:jar:1.1.5:compile
```

`3.6.8`:
```
[INFO] +- io.quarkus:quarkus-smallrye-jwt:jar:3.6.8:compile
[INFO] |  +- io.smallrye:smallrye-jwt:jar:4.4.0:compile
[INFO] |  +- jakarta.annotation:jakarta.annotation-api:jar:2.1.1:compile
[INFO] |  +- io.quarkus:quarkus-security:jar:3.6.8:compile
[INFO] |  \- io.quarkus:quarkus-jsonp:jar:3.6.8:compile
[INFO] |     \- org.eclipse.parsson:parsson:jar:1.1.5:compile
```

**Comment by gsmet** [comment]
Thanks for taking the time to provide the information. I added an entry to the 3.6 migration guide: https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.6#quarkus-smallrye-jwt-and-reactive-routes .
